### PR TITLE
Adds checks if a component name with same name exists

### PR DESCRIPTION
**Pull request by mik-dass**
Second part of #217 

note from @kadel:
https://github.com/redhat-developer/ocdev/pull/301 has to be merged before this

### CODE REVIEW

**Comment by cdrage** [comment]
@kadel @mik-dass I'm assuming this replaces: https://github.com/redhat-developer/ocdev/pull/263 ?

**Comment by kadel** [comment]
> @kadel @mik-dass I'm assuming this replaces: #263 ?

yes

**Comment by mik-dass** [comment]
@kadel Done

File: pkg/component/component.go [hunk 1]
```diff
@@ -104,6 +112,14 @@ func CreateFromDir(client *occlient.Client, name string, ctype string, dir strin
 		}
 	}
 
+	exists, err = Exists(client, name)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create git component %s", name)
+	}
+	if exists {
+		return errors.New("component with the name " + name + " already exists in the current project")
```
[the remark this answers is not in the document]
Comment (mik-dass) [reply pkg/component/component.go]: @kadel What should be fixed in this line except the current application thing?

**Comment by mik-dass** [comment]
@kadel Done

File: cmd/create.go [hunk 2]
```diff
@@ -58,6 +58,16 @@ If component name is not provided, component type value will be used for name.
 			os.Exit(1)
 		}
 
+		exists, err := component.Exists(client, componentName)
+		if err != nil {
+			checkError(err, "")
+			os.Exit(1)
```
Comment (kadel) [inline cmd/create.go]: you don't have to call `os.Exit(1)` after `checkError` it calls that

**Comment by mik-dass** [comment]
@kadel Done

**Comment by cdrage** [comment]
lgtm

**Pull request merged**